src/full/Agda/Auto/Auto.hs

```diff
@@ -350,15 +350,15 @@ auto ii rng argstr = liftTCM $ locallyTC eMakeCase (const True) $ do
                                          ) exprs
                       let msgs = catMaybes $ msg : map snd giveress
                           msg' = unlines msgs <$ guard (not $ null msgs)
-                      return $ AutoResult (Solutions $ catMaybes $ map fst giveress) msg'
+                      return $ AutoResult (Solutions $ mapMaybe fst giveress) msg'
 
      MCaseSplit -> do
       case thisdefinfo of
        Just (def, clause, True) ->
         case Map.elems tccons of
          [(m, mytype, mylocalVars, _)] | null eqcons -> do
           (ids, pats) <- constructPats cmap mi clause
-          let ctx = map (\((hid, id), t) -> HI hid (id, t)) (zip ids mylocalVars)
+          let ctx = zipWith (curry (\((hid, id), t) -> HI hid (id, t))) ids mylocalVars
```

Comment:
> Thank you for your effort in going through all of the changes! Overall I really think this is a big improvement over the old versions. I do have some nitpicks, and two more serious (and related) comments:
> 
> * Your refactorings sometimes require parentheses to be put around a whole do block, which makes them very hard to read. In general, I think a good rule is that parenthesis should always be closed on the same line.
> * You replaced several of my multiway `if` statements with regular `if/then/else`s. One of the reasons I prefer the multiway `if` (apart from a personal preference for them) is that they can be used to avoid the need for parentheses around the `do` block. So I think we should replace more `if/then/else` with multiway `if`s, not the other way around.

I don't disagree with your observations.  I was mainly seeing what could be achieved mechanically.  The .hlint.yaml file can be adjusted to pick what styles you want flagged and which should be ignored.  If I'd rehlinted, it would have flagged extra currying noise, I expect.  Left to my own devices I would also get rid of a lot of the redundant brackets and $s, and format the code with a mechanical formatter such as brittany, but I expect that that would upset those that like hand-crafted layout.  I will also have a look at the various warnings that ghc produces, but that I haven't looked into.